src/rpc/claimtrie.cpp

```diff
@@ -46,55 +48,54 @@ UniValue getclaimsintrie(const UniValue& params, bool fHelp)
             "      }\n"
             "    ]\n"
             "  }\n"
-            "]\n"
-        );
-    
+            "]\n");
+
     LOCK(cs_main);
-    UniValue ret(UniValue::VARR);
 
-    CCoinsViewCache view(pcoinsTip);
-    std::vector<namedNodeType> nodes = pclaimTrie->flattenTrie();
+    int maxHeight = chainActive.Height();
+    if (params.size() > 0)
+        maxHeight -= params[0].get_int() - 1;
 
+    CCoinsViewCache coins(pcoinsTip);
+    CClaimTrieCache claimTrie(pclaimTrie);
+    if (maxHeight < chainActive.Height())
+        if (!RollBackTo(coins, claimTrie, chainActive[maxHeight]))
```

Comment:
This method uses the approach of rolling back the caches to put us in a state where we were some number of blocks back. The caches are never flushed.